valve-fsm.txt

```diff
@@ -50,18 +65,46 @@ enum valve_state valve_fsm(enum valve_state current_state, enum valve_event even
     }
     return current_state;
 }
-#include <stdio.h>
-#include <unistd.h>
-#include <stbool.h>
-
-//booleans that light up panelboard indicator lights
-
-bool light_closed = false;
-bool light_opened = false;
-bool light_transition = false;
-bool light_failedtoopen = false;
-bool light_failedtoclose = false;
 
+void ind_lights(int valve_state)
```

Comment:
I would change the light_ values into a U8_t and use bitfield, where each bit is for a particular LED. Then you can just do lights &= valve_state to set it.

Reply:
actually lights = (1 << valve_state)

Reply:
did my best to implement this but I am completely out of my depth with bitwise operators.

Reply:
1. You don't need to use actual bitfields. I kind of hate them as their ordering is not guaranteed by the spec and is compiler dependent.
2. What I would do in this case, is use #defines 
```suggestion
#define LED_CLOSED 0
#define LED_OPEN 1
```

These define a bit position. Then you can have one variable and set it this way:
```suggestion
U8_t led = (1 << LED_CLOSED);
```

For the future, if you wanted a situation where you just wanted to toggle one LED and leave the rest in whatever state they are already in: 
```suggestion
led |= (1 << LED_CLOSED)  // Turn on LED CLOSED
led &= ~(1 << LED_CLOSED)  // Turn off LED CLOSED
```

Reply:
I don't know what kind of hardware you eventually want to run this on, but generally this is how the registers which control LEDs will be laid out in a processor. A 8-32 bit value in memory somewhere with individual bits determine the state of each of the LEDs